botocore/operation.py

```diff
@@ -86,8 +113,25 @@ def call(self, endpoint, **kwargs):
         self.session.emit(event, endpoint=endpoint,
                           model=self.model,
                           params=request_dict,
-                          operation=self)
+                          operation=self,
+                          request_signer=signer)
+
+        # This is kind of ugly, but it lets us hook into the request
+        # created event on a per-operation basis without mucking with
+        # the Endpoint/EndpointCreator objects.
+        real_emit = event_emitter.emit
+        def fake_emit(name, **kwargs):
+            if name.startswith('request-created'):
```

Comment:
I'm updating this now to register for the specific request-created.service.operation method, sign in a threadsafe fasion, do the request(s), then unregister. I think it's a nicer solution than the monkey patch. Thanks!